test/node/bson_corpus.spec.test.js

```diff
@@ -165,6 +165,11 @@ describe('BSON Corpus', function () {
         describe('valid-extjson', function () {
```

Comment:
it's unclear to me why the changes in this file are necessary.  reverting to what's currently in `main` doesn't make any tests fail.  could you explain why we're modifying the test runner?

Reply:
With toFixed(1) reintroduced we don't need to skip the multi-type test anymore, I put that back, but it is senitive to changes in string format as opposed to value equivalence. 

The changes here are meant to raise issue in the future if there's a change to the output format that would break round tripping the string into a numeric value for all doubles. Since the if stmt on main filtered only for the `'1.2345678921232E+18'` case we did not need the extra handling for the NaN and -0 cases. The second section which tests round tripping through BSON needs to handle the cases where EJSON automatically converts values to int/long/double based on size and fraction.